var/spack/repos/builtin/packages/python/package.py

```diff
@@ -260,7 +260,10 @@ class Python(Package):
     variant("bz2", default=True, description="Build bz2 module")
     variant("lzma", default=True, description="Build lzma module")
     variant("pyexpat", default=True, description="Build pyexpat module")
-    variant("ctypes", default=True, description="Build ctypes module")
+    variant("ctypes", default=True, description="Build ctypes module", when="platform=win32")
+    variant(
+        "external-libffi", default=True, description="Use external libffi instead of vendored one"
```

Comment:
We should never be using vendored copies of software if at all possible. In the case of libffi, we already have multiple patches for ppc64le and compiler support. We would have to add all of those same patches to python if we wanted to support their vendored libffi.

Reply:
Would it be better to enforce this by requiring `libffi` as a dependency unconditionally instead of having a variant that defaults to true?

Reply:
Depends on how complicated you want to make things. Personally, I wouldn't treat Windows differently than macOS/Linux unless we have to.

Reply:
Okay, I'd prefer to require `libffi` as a dependency unconditionally then as this would prevent people from building older versions of python with the vendored version and running into issues.